### PR TITLE
feat: build API change password

### PR DESCRIPTION
**Pull request by nhattrinhle**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/Code-Engine-Studio/ces-semantic-release/blob/main/README.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -1,6 +1,15 @@
 const { Created, OK } = require('../core/success.response')
 const { userService } = require('../services')
 
+const changePassword = async (req, res) => {
+    const userId = req.user?.userId
+    const { currentPassword, newPassword } = req.body
+    await userService.changePassword({ userId, currentPassword, newPassword })
+    new OK({
+        message: 'Change password success!'
```
Comment (ces-khanhhoang) [inline src/controllers/user.controller.js]: Message: Your password has been changed successfully.

File: src/services/user.service.js [hunk 2]
```diff
@@ -4,6 +4,38 @@ const { userRepo, tokenRepo } = require('../models/repo')
...
+    if (!user) {
+        throw new NotFoundError('Not found user')
+    }
+
+    const isMatchPassword = await bcrypt.compare(currentPassword, user.password)
+    if (!isMatchPassword) {
+        throw new AuthFailureError('Incorrect current password')
+    }
+    if (newPassword === currentPassword) {
+        throw new BadRequestError('The new password cannot be the same as the old one')
```
Comment (ces-khanhhoang) [inline src/services/user.service.js]: "New Password cannot be same as your current password. Please choose a different password."

File: src/controllers/user.controller.js (hunk 1 above)
Comment (nhattrinhle) [reply src/controllers/user.controller.js]: done

File: src/services/user.service.js (hunk 2 above)
Comment (nhattrinhle) [reply src/services/user.service.js]: done

**Review by ces-khanhhoang — APPROVED** [review]

**Pull request merged**